soccer/motion/MotionControl.cpp

```diff
@@ -178,45 +175,40 @@ void MotionControl::run() {
 
 
 		//	evaluate path - where should we be right now?
-		Point targetPos;
-		bool pathValidNow = _robot->path()->evaluate(
-			timeIntoPath,
-			targetPos,
-			targetVel);
+		bool pathValidNow = _robot->path()->evaluate(timeIntoPath, target);
 		if (!pathValidNow) {
-			targetVel.x = 0;
-			targetVel.y = 0;
+			target.vel = Geometry2d::Point();
 		}
 		//	tracking error
-		Point posError = targetPos - _robot->pos;
+		Point posError = target.pos - _robot->pos;
 
 		//	acceleration factor
-		Point nextTargetVel, _;
-		_robot->path()->evaluate(timeIntoPath + 1.0/60.0, _, nextTargetVel);
-		Point acceleration = (nextTargetVel - targetVel) / 60.0f;
+		Planning::MotionInstant nextTarget;
+		_robot->path()->evaluate(timeIntoPath + 1.0 / 60.0, nextTarget);
+		Point acceleration = (nextTarget.vel - target.vel) / 60.0f;
```

Comment:
No its fine I think. Everything gets rotated at the end.
But looking at this, it doesn't really make sense what its doing with the accel factor and stuff....
Also we're doing two path evaluates each time (also the path evaluate in the replanifneeded....)
This pathplanning and motion control stuff really needs to be rewritten.